src/generic/datavgen.cpp

```diff
@@ -631,6 +633,101 @@ class wxDataViewTreeNode
 
 WX_DEFINE_SORTED_ARRAY_SIZE_T(unsigned int, wxDataViewSelection);
 
+
+struct RangeEntity
+{
+    unsigned int from;
+    unsigned int to;
+};
+
+WX_DECLARE_OBJARRAY(RangeEntity, ArrayOfRangeEntities);
+#include "wx/arrimpl.cpp"
+WX_DEFINE_OBJARRAY(ArrayOfRangeEntities)
+
+class RowRange
+{
+public:
+    void Add(const unsigned int idx);
+    void RemoveFrom(const unsigned int idx);
+    bool Has(const unsigned int idx);
+    unsigned int Count();
+    unsigned int CountTo(const unsigned int idx);
```

Comment:
Having a comment describing this method would be helpful, if only to indicate whether `idx` is inclusive or exclusive here.

And it should be `const` too.